Game4.py

```diff
@@ -0,0 +1,165 @@
+import os, sys
+import pygame
+from random import randint
+
+from misc import *
+
+
+class Game4Main:
+    """The Main Game4 Class - This class handles the main
+    initialization and creating of the Game."""
+
+    def __init__(self, width=480,height=360):
+        # initialize pygame
+        pygame.init()
+        # set window size
+        self.width = width
+        self.height = height
+        # create screen
+        self.screen = pygame.display.set_mode((self.width, self.height))
+        # create clock
+        self.clock = pygame.time.Clock()
+        self.score = 0
+
+
+    def mainLoop(self):
+        """Main screen for game."""
+        # initialize player
+        count = 0
+        play_len = 25
+        play_x = 100
+        play_y = self.height - play_len
+        player = Player(play_x, play_y, play_len, self.screen)
+
+        # initialize stamina bars
+        P1_stamina_bar = StaminaBar(self.screen,25,"WHITE")
+        P2_stamina_bar = StaminaBar(self.screen,350,"RED")
+
+        # initialize obstacle length, x and y coordinates
+        obs_len = 25
+        obs_x = self.width
+        obs_y = self.height - obs_len
+        obstical_height = self.height - obs_len
+
+        # create list of obstacles
+        obstacles = []
+        new_obstical = False
+        # uncomment and place line below in [] of obstacles = [] to generate random obstacles:
+        # Obstacle(obs_x, obs_y, obs_len, self.screen,'BLUE')
+
+
+        # initialize time variables
+        prev_time = 0
+        obs_dt = 500
+
+        # main event loop
+        while 1:
+            count+=1
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    sys.exit()
+                # check keyboard for obstacle player
+                if event.type == pygame.KEYDOWN:
+                    # z to create obstacle at ground level
+                    if event.key == pygame.K_z:
+                        new_obstical = True
```

Comment:
sp. `new_obstacle`

Reply:
Consider using constants to document the levels:

```python
GROUND_LEVEL_X_OFFSET = 0
FIRST_LEVEL_X_OFFSET = 25
SECOND_LEVEL_X_OFFSET = 50
…
    obstacle_height = obs_y - FIRST_LEVEL_X_OFFSET
```

The general principle is that you can name things instead of commenting them. The names are more likely to stay in sync with the code that the comments are, and are also evident everywhere the named thing is used.

You could also use e.g.

```python
LEVEL_OFFSETS = {
   'ground': 0,
  'first': 25,
  'second': 50,
}
…
    obstacle_height = obs_y - LEVEL_OFFSETS['first']
```

or:

```python
LEVEL_OFFSETS = {
  0: 0,
  1: 25,
  2: 50,
}
…
    obstacle_height = obs_y - LEVEL_OFFSETS[1]
```


```python
LEVEL_OFFSETS = [0, 25, 50]
…
    obstacle_height = obs_y - LEVEL_OFFSETS[1]
```

This makes it easier to add levels. It also allows for downstream simplifications, for example a map from event.key to level name or index.